File: GXpert/GXpert.Web/Modules/ServerTypes/Attendance/StudentClassAttendanceForm.ts
import { ServiceLookupEditor, LookupEditor, EnumEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { EAttendanceStatus } from "../Web/Enums.EAttendanceStatus";

export interface StudentClassAttendanceForm {
    InstituteTimeTableId: ServiceLookupEditor;
    StudentId: LookupEditor;
    AttendanceStatus: EnumEditor;
}

export class StudentClassAttendanceForm extends PrefixedContext {
    static readonly formKey = 'Attendance.StudentClassAttendance';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!StudentClassAttendanceForm.init)  {
            StudentClassAttendanceForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = LookupEditor;
            var w2 = EnumEditor;

            initFormType(StudentClassAttendanceForm, [
                'InstituteTimeTableId', w0,
                'StudentId', w1,
                'AttendanceStatus', w2
            ]);
        }
    }
}

[EAttendanceStatus]; // referenced types

File: GXpert/GXpert.Web/Modules/Attendance/StudentClassAttendance/StudentClassAttendanceForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Attendance.Forms;

[FormScript("Attendance.StudentClassAttendance")]
[BasedOnRow(typeof(StudentClassAttendanceRow), CheckNames = true)]
public class StudentClassAttendanceForm
{
    [HalfWidth]
    public int InstituteTimeTableId { get; set; }
    [HalfWidth]
    public int StudentId { get; set; }
    [HalfWidth]
    public short AttendanceStatus { get; set; }
   
}